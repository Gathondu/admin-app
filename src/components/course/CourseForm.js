import React from 'react';
import PropTypes from 'prop-types';
import TextInput from '../common/TextInput'
import SelectInput from '../common/SelectInput'

const CourseForm = ({course, allAuthors, onSave, onChange, loading, errors, formIsValid}) => {
    return (
        <form>
            <h1>Manage Course</h1>
            <TextInput
                label="Title"
                onChange={onChange}
                name="title"
                value={course.title}
                error={errors.title}
                placeholder="Enter Course Title"/>
            <SelectInput
                label="Author"
                onChange={onChange}
                name="authorId"
                value={course.authorId}
                defaultOption="Select Author"
                options={allAuthors}
                error={errors.authorId}/>
            <TextInput
                label="Category"
                onChange={onChange}
                name="category"
                value={course.category}
                error={errors.category}
                placeholder="Enter Course Category"/>
            <TextInput
                label="Length"
                onChange={onChange}
                name="length"
                value={course.length}
                error={errors.length}
                placeholder="Enter Course Length"/>
            <input
                type="submit"
                disabled={loading || !formIsValid}
                value={loading ? 'Saving...' : 'Save'}
                className="btn btn-primary"
                onClick={onSave}/>
        </form>
    );
};

CourseForm.propTypes = {
    course: PropTypes.object.isRequired,
    allAuthors: PropTypes.array,
    onSave: PropTypes.func.isRequired,
    onChange: PropTypes.func.isRequired,
    loading: PropTypes.bool,
    errors: PropTypes.object
};
export default CourseForm;
