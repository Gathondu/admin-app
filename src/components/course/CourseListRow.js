import React from 'react';
import PropTypes from 'prop-types';
import {Link} from "react-router-dom";

const CourseListRow = ({course, deleteCourse}) => {
    return (
        <tr>
            <td><i className="fa fa-trash" onClick={() => deleteCourse(course.id)}/></td>
            <td><a href={course.watchHref} target="_blank" rel="noopener noreferrer"><i className="fa fa-television"/></a></td>
            <td><Link to={`/course/${course.id}`}>{course.title}</Link></td>
            <td>{course.authorId}</td>
            <td>{course.category}</td>
            <td>{course.length}</td>
        </tr>
    );
};

CourseListRow.propTypes = {
    course: PropTypes.object.isRequired
};

export default CourseListRow;
