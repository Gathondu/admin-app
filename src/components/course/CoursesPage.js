import React, {Component} from 'react';
import PropTypes from 'prop-types';
import AddCourseForm from "./AddCourseForm";
import Course from './Course';

class CoursesPage extends Component {
    render() {
        const {
            courses,
            addCourse,
            removeCourse
        } = this.props;
        const courseComponents = courses.map((course, index) => (
            <Course
                title={course.title}
                description={course.description}
                key={index}
                index={index}
                removeCourse={removeCourse}
            />
        ));
        return (
            [
                <h1>{this.props.title}</h1>,
                [courseComponents],
                <AddCourseForm addCourse={addCourse}/>]
        );
    }
}

CoursesPage.propTypes = {
    title: PropTypes.string.isRequired,
    courses:PropTypes.array.isRequired,
    addCourse: PropTypes.func.isRequired,
    removeCourse: PropTypes.func.isRequired
};

export default CoursesPage;
