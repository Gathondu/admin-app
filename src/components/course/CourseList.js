import React from 'react';
import PropTypes from 'prop-types';
import CourseListRow from "./CourseListRow";

const CourseList = ({courses, removeCourse}) => {
    const courseRows = courses.map((course, index) => (
        <CourseListRow
            course={course}
            key={course.id}
            index={index}
            removeCourse={removeCourse}
        />
    ));
    return (
        <table className="table">
            <thead>
                <tr>
                    <th>&nbsp;</th>
                    <th>Title</th>
                    <th>Author</th>
                    <th>Category</th>
                    <th>Length</th>
                </tr>
            </thead>
            <tbody>{courseRows}</tbody>
        </table>
    );
};

CourseList.propTypes = {
    courses: PropTypes.array.isRequired,
    removeCourse: PropTypes.func.isRequired
};

export default CourseList;
