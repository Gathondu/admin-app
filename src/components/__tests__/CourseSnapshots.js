import renderer from 'react-test-renderer';
import React from 'react';
import {BrowserRouter} from "react-router-dom";
import CourseForm from '../course/CourseForm';
import CourseList from '../course/CourseList';
import CourseListRow from '../course/CourseListRow';
import CoursesPage from '../course/CoursesPage';

describe('Course Components',()=>{
    describe('CourseForm Component', () => {
        it('should not regress', () => {
            const tree = renderer.create(
                <CourseForm onChange={()=>{}} onSave={()=>{}} course={{}} errors={{}} allAuthors={[]}/>
            ).toJSON();
            expect(tree).toMatchSnapshot();
        });
    });
    describe('CourseListRow Component', () => {
        it('should not regress', () => {
            const tree = renderer.create(
                <BrowserRouter>
                    <CourseListRow deleteCourse={()=>{}} course={{}}/>
                </BrowserRouter>
            ).toJSON();
            expect(tree).toMatchSnapshot();
        });
    });
    describe('CourseList Component', () => {
        it('should not regress', () => {
            const tree = renderer.create(
                <CourseList deleteCourse={()=>{}} courses={[]} loading={false}/>
            ).toJSON();
            expect(tree).toMatchSnapshot();
        });
    });
    describe('CoursesPage Component', () => {
        it('should not regress', () => {
            const tree = renderer.create(
                <CoursesPage deleteCourse={()=>{}} courses={[]} title={'test courses'} loading={false}/>
            ).toJSON();
            expect(tree).toMatchSnapshot();
        });
    });
});