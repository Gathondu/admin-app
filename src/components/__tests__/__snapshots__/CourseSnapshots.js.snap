// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Course Components CourseForm Component should not regress 1`] = `
<form>
  <h1>
    Manage Course
  </h1>
  <div
    className="form-group"
  >
    <label
      htmlFor="title"
    >
      Title
    </label>
    <div
      className="field"
    >
      <input
        className="form-control"
        name="title"
        onChange={[Function]}
        type="text"
      />
    </div>
  </div>
  <div
    className="form-group"
  >
    <label
      htmlFor="authorId"
    >
      Author
    </label>
    <div
      className="field"
    >
      <select
        className="form-control"
        name="authorId"
        onChange={[Function]}
      >
        <option
          value=""
        >
          Select Author
        </option>
      </select>
    </div>
  </div>
  <div
    className="form-group"
  >
    <label
      htmlFor="category"
    >
      Category
    </label>
    <div
      className="field"
    >
      <input
        className="form-control"
        name="category"
        onChange={[Function]}
        type="text"
      />
    </div>
  </div>
  <div
    className="form-group"
  >
    <label
      htmlFor="length"
    >
      Length
    </label>
    <div
      className="field"
    >
      <input
        className="form-control"
        name="length"
        onChange={[Function]}
        type="text"
      />
    </div>
  </div>
  <input
    className="btn btn-primary"
    disabled={true}
    onClick={[Function]}
    type="submit"
    value="Save"
  />
</form>
`;

exports[`Course Components CourseList Component should not regress 1`] = `
<div
  className="jumbotron"
>
  <i
    aria-hidden="true"
    className="fa fa-ban fa-5x"
  />
  <h2>
    No Courses to display. Add course to view
  </h2>
</div>
`;

exports[`Course Components CourseListRow Component should not regress 1`] = `
<tr>
  <td>
    <i
      className="fa fa-trash"
      onClick={[Function]}
    />
  </td>
  <td>
    <a
      rel="noopener noreferrer"
      target="_blank"
    >
      <i
        className="fa fa-television"
      />
    </a>
  </td>
  <td>
    <a
      href="/course/undefined"
      onClick={[Function]}
    />
  </td>
  <td />
  <td />
  <td />
</tr>
`;

exports[`Course Components CoursesPage Component should not regress 1`] = `
Array [
  <h1>
    test courses
  </h1>,
  <input
    className="btn btn-primary"
    onClick={[Function]}
    type="submit"
    value="Add Course"
  />,
  <div
    className="jumbotron"
  >
    <i
      aria-hidden="true"
      className="fa fa-ban fa-5x"
    />
    <h2>
      No Courses to display. Add course to view
    </h2>
  </div>,
]
`;
